### ISSUE TITLE
Unable to attach DNS tcp monitor to DNS VirtualServer

### ISSUE THREAD

**Issue by chen23** [F5Networks/f5-appsvcs-extension#160]
##### Do you already have an issue opened with F5 support?
Github Issues are consistently monitored by F5 staff, but should be considered as best effort only and you should not expect to receive the same level of response as provided by F5 Support. Please [open an case](https://support.f5.com/csp/article/K2633) with F5 if this is a critical issue.

##### ISSUE TYPE

 - Bug Report

##### AS3 BUILD/ VERSION

```
{
  "release": "4",
  "schemaCurrent": "3.14.0",
  "schemaMinimum": "3.0.0",
  "version": "3.14.0"
}
```

##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     13.1.1.2
  Build       0.0.4
  Edition     Point Release 2
  Date        Thu Oct 11 15:32:21 PDT 2018
```

##### SUMMARY

error when you try and attach a "tcp" monitor to a DNS Virtual Server

##### STEPS TO REPRODUCE

sample post

```
{
  "class": "ADC",
  "schemaVersion": "3.9.0",
  "Common": {
    "class": "Tenant",
    "Shared": {
      "class": "Application",
      "template": "shared",
      "AS3DataCenter": {
        "class": "GSLB_Data_Center"
      },
      "AS3Server": {
        "class": "GSLB_Server",
        "dataCenter": {
          "use": "AS3DataCenter"
        },
        "devices": [
          {
            "address": "10.1.10.240",
            "addressTranslation": "10.1.10.240"
          }
        ],
        "virtualServers": [
          {
            "address": "10.1.10.11",
            "addressTranslation": "10.1.10.11",
            "addressTranslationPort": 443,
            "port": 443,
            "monitors": [
              {
                "bigip": "/Common/tcp"
              }
            ]
          }
        ]
      }
    }
  }
}
```

##### EXPECTED RESULTS

tcp monitor should be attached.
##### ACTUAL RESULTS

Error.  Seems OK for "http" and "https"

```
{"code":422,"declarationFullId":"","message":"Unable to find /Common/tcp for /Common/Shared/AS3Server/virtualServers/0/monitors/0"}
```

**Comment by chen23** [comment]
same error for gateway_icmp
```
{"code":422,"declarationFullId":"","message":"Unable to find /Common/gateway_icmp for /Common/Shared/AS3Server/virtualServers/0/monitors/0"}
```

**Comment by dstokesf5** [comment]
It looks like only bigip, http, and https are currently supported in the schema. I have created internal product backlog item AUTOTOOL-973 to track adding the remaining monitor types.

**Comment by dstokesf5** [comment]
This has been resolved as part of the 3.20.0 release.
